packages/utils/upgrade/resources/codemods/5.0.0/comment-out-lifecycle-files.code.ts

```diff
@@ -0,0 +1,63 @@
+import { Transform } from 'jscodeshift';
+
+/**
+ * comments out lifecycles.js/ts files and adds a description for the reason at the top
+ */
+const transform: Transform = (file, api) => {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+
+  // check if file path follows this pattern `content-types/[content-type-name]/lifecycles`
+  if (/content-types\/[^/]+\/lifecycles\.(js|ts)$/.test(file.path)) {
```

Comment:
It could've been easier to maintain to assert the path validity using glob, but that'll do. Does the path has to start with content-types? Or can it be `src/api/foo/content-types`? Will the regular expression work for this?

Reply:
`src/api/foo/content-types/foo/lifecycles.js/ts` will work for this yes, it'll match the regex, this is already the case if you test the codemod